zim/plugins/base/imagegenerator.py

```diff
@@ -183,6 +183,18 @@ def check_image_file(new_script_file):
 			self.image_file = _stitch_fileextension(self.script_file, imagefile_extension)
 			self.attrib['src'] = './' + self.image_file.basename
 
+		if self.script_file.exists() and (
+				not self.image_file.exists() or
+				self.script_file.mtime() > self.image_file.mtime()):
```

Comment:
Might want to consider `+ 1` in case of racey behavior on some filesystems: https://stackoverflow.com/questions/8246933/detect-mtime-resolution-of-filesystem